state/allwatcher.go

```diff
@@ -119,6 +119,11 @@ type backingModel modelDoc
 
 func (e *backingModel) updated(st *State, store *multiwatcherStore, id string) error {
 	cfg, err := st.ModelConfig()
+	// Treat a not found error here as if the model has been removed.
+	// This will be the case if the model has been marked dead, pending cleanup.
+	if errors.IsNotFound(err) && e.Life == Dead {
```

Comment:
make a helper for this? Then the comment doesn't need to be repeated each time either. Something like `ifNotFoundAndModelDead(err, e)`